Content.IntegrationTests/Tests/Inventory/ClothingEquipTest.cs

```diff
@@ -0,0 +1,176 @@
+    protected override string PlayerPrototype => "MobHuman";
+
+    [Test]
+    public async Task EquipAllClothingTesting()
+    {
+        var clothingProtos = ProtoMan.EnumeratePrototypes<EntityPrototype>()
+            .Where(x => x.HasComponent<ClothingComponent>()
+                        && !x.HideSpawnMenu
+                        && !x.Abstract
+                        && !Pair.IsTestPrototype(x)
+                        && !x.HasComponent<RandomSpawnerComponent>()
+                        // Mobs are a massive pain in this test
+                        // (Admeme mouse eating your cak, cancer mouse killing you... mice just dying...)
+                        && !x.HasComponent<MobStateComponent>());
+
+        List<EntityUid> clothings = [];
+        EntityUid jumpsuit = default!;
+        EntityUid hardsuit = default!;
+        await Server.WaitPost(() =>
+        {
+            foreach (var clothing in clothingProtos)
+            {
+                var ent = SEntMan.SpawnAtPosition(clothing.ID, ToServer(PlayerCoords));
+                clothings.Add(ent);
+
+                // Material bag can pick up other clothing items.
+                SEntMan.RemoveComponent<MagnetPickupComponent>(ent);
+            }
+
+            jumpsuit = SEntMan.SpawnAtPosition(JumpsuitProto, ToServer(PlayerCoords));
+            hardsuit = SEntMan.SpawnAtPosition(HardsuitProto, ToServer(PlayerCoords));
+        });
+
+        await AddAtmosphere();
+
+        var invSystem = Server.System<ServerInventorySystem>();
+        var invComp = SEntMan.GetComponent<InventoryComponent>(SPlayer);
+
+        var actionSys = Server.System<ActionsSystem>();
+        var baseActions = actionSys.GetActions(SPlayer).ToList();
+
+        foreach (var clothing in clothings)
+        {
+            Assert.That(SEntMan.TryGetComponent<ClothingComponent>(clothing, out var clothingComp),
+                $"{SEntMan.ToPrettyString(clothing)} doesn't have ClothingComponent");
+            foreach (var slot in invComp.Slots)
+            {
+                if ((slot.SlotFlags & clothingComp!.Slots) == 0x0)
+                    continue;
+
+                await Server.WaitAssertion(() =>
+                {
+                    if (slot.SlotFlags is SlotFlags.IDCARD or SlotFlags.POCKET)
+                        DoEquip(invSystem, jumpsuit, "jumpsuit");
+
+                    if (slot.SlotFlags == SlotFlags.SUITSTORAGE)
+                        DoEquip(invSystem, hardsuit, "outerClothing");
+
+                    DoEquip(invSystem, clothing, slot.Name, clothingComp);
+                });
+                await RunTicks(1);
+
+                var newActions = actionSys.GetActions(SPlayer)
+                    .Except(baseActions)
+                    // Don't bother with more complicated actions (they can take targets, etc.)
+                    .Where(action => SEntMan.HasComponent<InstantActionComponent>(action.Owner))
+                    .ToList();
```

Comment:
While this does end up passing on both here and a downstream I tried the test on, this feels like it's asking for trouble. Consider an item that grants an instant action which scrambles you. That would then put the player out of range of other items, killing the test.

Should this be filtered down to just actions that have a ToggleAction component? Or even just ToggleClothing?